CRM/Core/Form.php

```diff
@@ -2243,7 +2243,7 @@ protected function setContactID() {
       return (int) $tempID;
     }
 
-    $userID = $this->getLoggedInUserContactID();
+    $userID = CRM_Core_Session::getLoggedInContactID();
 
     if (!is_null($tempID) && $tempID === $userID) {
```

Comment:
The only difference I can see is that we are changing from a function that returns FALSE to one that returns NULL. So this is_null check maybe needs to become is false (I'm struggling a bit to figure out what it achieves. I know that 0 and potentially '0' are valid / meaningful in this function but I'm not sure the meaning of FALSE - ie could $tempID ever be FALSE & what would that mean)

Reply:
May be if ```CRM_Core_Session::getLoggedInContactID``` return int|null will resolve the conflict here.

Reply:
@eileenmcnaughton In this case $tempID is set by CRM_Utils_Request::retrieve and will be NULL if not found. It then may be set via ['select_contact_id'] but only if that is not empty (so not NULL). So the `is_null` check never changes because $tempID is never set by `getLoggedInContactID()`

When we do the `$tempID === $userID` we already checked that $tempID is not null (via `!is_null()`) so we don't need to worry what the value of `$userID` is - if it is NULL or FALSE it won't match because we know that `$tempID` is not NULL and it cannot be set to FALSE in the function